oauthproxy.go

```diff
@@ -949,86 +992,20 @@ func (p *OAuthProxy) Proxy(rw http.ResponseWriter, req *http.Request) {
 // Set-Cookie headers may be set on the response as a side-effect of calling this method.
 func (p *OAuthProxy) getAuthenticatedSession(rw http.ResponseWriter, req *http.Request) (*sessionsapi.SessionState, error) {
 	var session *sessionsapi.SessionState
-	var err error
-	var saveSession, clearSession, revalidated bool
 
-	if p.skipJwtBearerTokens && req.Header.Get("Authorization") != "" {
-		session, err = p.GetJwtSession(req)
-		if err != nil {
-			logger.Printf("Error retrieving session from token in Authorization header: %s", err)
-		}
-		if session != nil {
-			saveSession = false
-		}
-	}
+	getSession := p.sessionChain.Then(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		session = middleware.GetRequestScope(req).Session
+	}))
+	getSession.ServeHTTP(rw, req)
 
-	remoteAddr := ip.GetClientString(p.realClientIPParser, req, true)
 	if session == nil {
-		session, err = p.LoadCookiedSession(req)
-		if err != nil {
-			logger.Printf("Error loading cookied session: %s", err)
-		}
-
-		if session != nil {
-			if session.Age() > p.CookieRefresh && p.CookieRefresh != time.Duration(0) {
-				logger.Printf("Refreshing %s old session cookie for %s (refresh after %s)", session.Age(), session, p.CookieRefresh)
-				saveSession = true
-			}
-
-			if ok, err := p.provider.RefreshSessionIfNeeded(req.Context(), session); err != nil {
-				logger.Printf("%s removing session. error refreshing access token %s %s", remoteAddr, err, session)
-				clearSession = true
-				session = nil
-			} else if ok {
-				saveSession = true
-				revalidated = true
-			}
-		}
-	}
-
-	if session != nil && session.IsExpired() {
-		logger.Printf("Removing session: token expired %s", session)
-		session = nil
-		saveSession = false
-		clearSession = true
-	}
-
-	if saveSession && !revalidated && session != nil && session.AccessToken != "" {
-		if !p.provider.ValidateSessionState(req.Context(), session) {
-			logger.Printf("Removing session: error validating %s", session)
-			saveSession = false
-			session = nil
-			clearSession = true
-		}
+		return nil, ErrNeedsLogin
 	}
 
 	if session != nil && session.Email != "" && !p.Validator(session.Email) {
```

Comment:
Unrelated to this PR -- do you think the global email domain validator that is required for all providers be moved in the structured configs (IE session authorization rules)?

I take it the `email-domains` validator's specific intent was for providers like Google, where is has little applicability to OIDC where users can likely tune the RBAC on the OIDC provider configuration side.

This is likely the reason hacky `session.Email = claim.Subject if claim.Email = nil` stuff has happened in Bearer JWT code paths? To make sure a email always exists for the mandatory email domain validator?

Reply:
>Unrelated to this PR -- do you think the global email domain validator that is required for all providers be moved in the structured configs (IE session authorization rules)?

Yeah precisely that, I imagine we will move that into the session authorization layer once we implement that 

>I take it the email-domains validator's specific intent was for providers like Google, where is has little applicability to OIDC where users can likely tune the RBAC on the OIDC provider configuration side.
>
>This is likely the reason hacky session.Email = claim.Subject if claim.Email = nil stuff has happened in Bearer JWT code paths? To make sure a email always exists for the mandatory email domain validator?

I assume so, though I'd have to do some archeology to confirm, was before my time!